lms/djangoapps/courseware/management/commands/dump_course_structure.py

```diff
@@ -165,30 +165,37 @@ def dump_module_by_position(course_id, module, level=0,
     else:
         display_name_long = prefix + "," + module.display_name
 
-    if destination == None:
-        destination = [ ('course_id',
-                         'position',
-                         'level',
-                         'module_id',
-                         'type',
-                         'displayname',
-                         'path',
-                         'parent') ]
-
-    destination.append( (course_id,
-                         pos, 
-                         level, 
-                         module.id, 
-                         module.location.category,
-                         module.display_name, 
-                         display_name_long,
-                         parent) )
+    if destination is None:
+        destination = [(
+            'course_id',
+            'position',
+            'level',
+            'module_id',
+            'type',
+            'displayname',
+            'path',
+            'parent')]
+
+    #  This was a bit confusing for me - was this the intent?
+    #  I think 'extend' is for appending a sequence...
+    #  ...which means we need to use parens to close the tuple?
```

Comment:
Yes, I think the intent is to use `append` and not `extend`.

Let's trace `destination` through...

```python
# If `destination` is not passed in, default to `None`.
def dump_module_by_position(..., destination=None, ...):
    if destination is None:
        # Initialize it.
        # The right-hand side starts and ends with square brackets `[]`, so it's a List.
        # The first item inside the List starts and ends with parens `()`, so it's a Tuple.
        # `destination` is a List of Tuples.
        # It's a bit easier to read at a glance if it's reformatted like:
        destination = [
            (
                'course_id',
                '...',
            ),
        ]
        # Since this is the default initialization,
        # I'm going to assume that if `destination` is passed in,
        # it's also a List of Tuples.
    # Right now, the code looks like this:
    destination.append(
        (
            course_id,
            '...',
        )
    )
    # So it appends a Tuple to the list
    print(destination)
    # [('course_id', '...'), ('course_id', '...')]

    # Simply swapping `extend` for `append`
    destination.extend(
        (
            course_id,
            '...',
        )
    )
    # would give us
    # [('course_id', '...'), 'course_id', '...']
    # which is no longer a a List of Tuples,
    # but rather a List with a Tuple and then non-grouped values.
    # If we _really_ wanted to use `extend` here, we would need to
    # construct a list of Tuples to pass in, like
    destination.extend(
        [
            (
                course_id,
                '...',
            )
        ]
    )
    # but that's needlessly complicated.
```

Let me know what you think.